draft-ietf-quic-transport.md

```diff
@@ -666,64 +666,226 @@ Version negotiation uses unprotected data.  The result of the negotiation MUST
+  {{?RFC7301}} for this purpose)
+
+* for the server, assurance that the client can receive packets that are
+  addressed with the transport address that is claimed by the client (see
+  {{source-address-token}})
+
+Details of how TLS is integrated with QUIC is provided in more detail in
+{{QUIC-TLS}}.
+
+
+## Transport Parameters
+
+During connection establishment, both endpoints make authenticated declarations
+of their transport parameters.  These declarations are made unilaterally by each
+endpoint.  Endpoints are required to comply with the restrictions implied by
+these parameters; the description of each parameter includes rules for its
+handling.
 
-### Transport Parameters and Options
+The format of the transport parameters is the TransportParameters struct from
+{{figure-transport-parameters}}.  This is described using the presentation
+language from Section 3 of {{!I-D.ietf-tls-tls13}}.
 
-During connection establishment, the handshake must negotiate various transport
-parameters.  The currently defined transport parameters are described later in
-the document.
+~~~
+   uint32 QuicVersion;
+
+   enum {
+      SFCW(0),
+      CFCW(1),
+      MSPC(2),
+      ICSL(3),
+      TCID(4),
+      COPT(5),
+      (65535)
+   } TransportParameterId;
+
+   struct {
+      TransportParameterId parameter;
+      opaque value<0..2^16-1>;
+   } TransportParameter;
+
+   struct {
+      select (Handshake.msg_type) {
+         case client_hello:
+            QuicVersion negotiated_version;
+            QuicVersion initial_version;
+
+         case encrypted_extensions:
+            QuicVersion supported_versions<2..2^8-4>;
+      };
+      TransportParameter parameters<30..2^16-1>;
+   } TransportParameters;
+~~~
+{: #figure-transport-parameters title="Definition of TransportParameters"}
+
+The "extension_data" field of the quic_transport_parameters extension defined in
+{{QUIC-TLS}} contains a TransportParameters value.  TLS encoding rules are
+therefore used to encode the transport parameters.
+
+Definitions for each of the defined transport parameters are included in
+{{transport-parameter-definitions}}.  Use of the version fields from transport
+parameters is described in {{version-validation}}.
+
+
+### Transport Parameter Definitions
+
+An endpoint MUST include the following parameters in its encoded
+TransportParameters:
 
-The transport component of the handshake is responsible for exchanging and
-negotiating the following parameters for a QUIC connection.  Not all parameters
-are negotiated, some are parameters sent in just one direction.  These
-parameters and options are encoded and handed off to the crypto handshake
-protocol to be transmitted to the peer.
+SFCW (0x0000):
 
-#### Encoding
+: The initial stream level flow control parameter is encoded as an unsigned
+  32-bit integer.  The sender of this parameter indicates that the flow control
+  offset for all stream data sent toward it is this value.
```

Comment:
This is per-stream. Perhaps something like: "The sender of this parameter indicates that the initial flow control offset for data sent toward it on any stream is this value."